src/Files.App/ViewModels/UserControls/OngoingTasksViewModel.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.App.ViewModels.UserControls
+{
+	public class OngoingTasksViewModel : ObservableObject, IOngoingTasksActions
+	{
+		public ObservableCollection<StatusBanner> StatusBannersSource { get; private set; } = new();
+
+		private int _MedianOperationProgressValue = 0;
+		public int MedianOperationProgressValue
+		{
+			get => _MedianOperationProgressValue;
+			private set => SetProperty(ref _MedianOperationProgressValue, value);
+		}
+
+		public int OngoingOperationsCount
+		{
+			get
+			{
+				int count = 0;
+
+				foreach (var item in StatusBannersSource)
+				{
+					if (item.IsProgressing)
+					{
+						count++;
+					}
+				}
+
+				return count;
+			}
+		}
+
+		public bool AnyOperationsOngoing
+			=> OngoingOperationsCount > 0;
+
+		public bool AnyBannersPresent
+			=> StatusBannersSource.Any();
+
+		public int InfoBadgeState
+		{
+			get
+			{
+				var anyFailure = StatusBannersSource.Any(i => i.Status != ReturnResult.InProgress && i.Status != ReturnResult.Success);
+
+				return (anyFailure, AnyOperationsOngoing) switch
+				{
+					(false, false) => 0, // All success
+					(false, true) => 1,  // Ongoing
+					(true, true) => 2,   // Ongoing with failure
+					(true, false) => 3   // Completed with failure
+				};
+			}
+		}
+
+		public int InfoBadgeValue
+			=> OngoingOperationsCount > 0 ? OngoingOperationsCount : -1;
+
+		public event EventHandler<PostedStatusBanner>? ProgressBannerPosted;
+
+		public OngoingTasksViewModel()
+		{
+			StatusBannersSource.CollectionChanged += (s, e) => OnPropertyChanged(nameof(AnyBannersPresent));
+		}
+
+		public PostedStatusBanner PostBanner(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation)
+		{
+			StatusBanner banner = new StatusBanner(message, title, initialProgress, status, operation);
+			PostedStatusBanner postedBanner = new PostedStatusBanner(banner, this);
+
+			StatusBannersSource.Insert(0, banner);
+			ProgressBannerPosted?.Invoke(this, postedBanner);
+
+			UpdateBanner(banner);
+
+			return postedBanner;
+		}
+
+		public PostedStatusBanner PostOperationBanner(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation, CancellationTokenSource cancellationTokenSource)
+		{
+			StatusBanner banner = new StatusBanner(message, title, initialProgress, status, operation)
+			{
+				CancellationTokenSource = cancellationTokenSource,
+			};
+
+			PostedStatusBanner postedBanner = new PostedStatusBanner(banner, this, cancellationTokenSource);
```

Comment:
You should use new() or var to avoid repetition

Reply:
Done